src/FondOfSpryker/Zed/CompanyUserCompanyAssigner/Communication/Plugin/CompanyUserExtension/CompanyUserCompanyAssignerCompanyUserPreSavePlugin.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace FondOfSpryker\Zed\CompanyUserCompanyAssigner\Communication\Plugin\CompanyUserExtension;
+
+use FondOfSpryker\Zed\CompanyUserCompanyAssigner\Dependency\CompanyUserCompanyAssignerEvents;
+use Generated\Shared\Transfer\CompanyRoleCollectionTransfer;
+use Generated\Shared\Transfer\CompanyUserResponseTransfer;
+use Generated\Shared\Transfer\CompanyUserTransfer;
+use Spryker\Zed\CompanyUserExtension\Dependency\Plugin\CompanyUserPreSavePluginInterface;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+
+/**
+ * @method \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Business\CompanyUserCompanyAssignerFacadeInterface getFacade()
+ * @method \FondOfSpryker\Zed\CompanyUserCompanyAssigner\Communication\CompanyUserCompanyAssignerCommunicationFactory getFactory()
+ * @method \FondOfSpryker\Zed\CompanyUserCompanyAssigner\CompanyUserCompanyAssignerConfig getConfig()
+ */
+class CompanyUserCompanyAssignerCompanyUserPreSavePlugin extends AbstractPlugin implements CompanyUserPreSavePluginInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\CompanyUserResponseTransfer $companyUserResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\CompanyUserResponseTransfer
+     */
+    public function preSave(CompanyUserResponseTransfer $companyUserResponseTransfer): CompanyUserResponseTransfer
+    {
+        $companyUserTransfer = $companyUserResponseTransfer->getCompanyUser();
+
+        if ($companyUserTransfer === null || $companyUserResponseTransfer->getIsSuccessful() !== true) {
+            return $companyUserResponseTransfer;
+        }
+
+        $companyUserRoleCollection = $this->getFacade()->getCompanyUserRoleCollection($companyUserTransfer);
+
+        if (!$this->isCompanyUserRolesChanged($companyUserRoleCollection, $companyUserTransfer)) {
```

Comment:
`haveCompanyUserRolesChanged` or `didCompanyUserRoleChange` would be more compelling imho